lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java

```diff
@@ -2007,6 +2007,44 @@ public void no_messages_when_channel_state_not_attached() {
         }
     }
 
+    /*
+     * Without creating a channel, send a DETACHED protocol message to a named channel.
+     *
+     * Assert that the channel is not created when processing the message and that, therefore,
+     * the message is dropped. This prevents issues where releasing a channel - dropping it from
+     * the channel map and calling detach, can cause the channel to be re-created when the
+     * DETATCHED response comes back from ably.
+     */
+    @Test
+    public void messages_to_non_existent_channels_are_dropped() throws AblyException {
```

Comment:
I think for this case it would also be useful to show that there was an attached channel that was detached - (with call to `detach()`) and then send the `detached()` message afterwards

Reply:
I'm a little confused as to what you want me to add to the test, or what that test case would be covering? This test aims to cover what should happen if a channel doesn't exist in the channel map, rather than what state the channel is in?

Reply:
I meant to replay the whole scenario where there is an attach and the map contains a channel at some point and then the detach causing it to disappear - then test the afterwards :)

Reply:
Ah got you - I have pushed a test that allows that :)